superset/security/analytics_db_safety.py

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+class DBSecurityException(Exception):
+    """ Exception to prevent a security issue with connecting a DB """
+
+    status = 400
+
+
+def check_sqlalchemy_uri(uri):
+    if uri.startswith("sqlite"):
```

Comment:
@suddjian should this be `if uri.drivername == "sqlite":`? Also could you add typing to this method so it's apparent the type of the `uri` method.

Reply:
This PR is merged but I can add typing in a new PR.

`uri` is a string so I assume you're referring to the output of `make_url` from sqlalchemy. `make_url(uri).drivername == "sqlite"` won't quite work in all cases because [there are actually multiple drivers](https://docs.sqlalchemy.org/en/13/dialects/sqlite.html#dialect-sqlite-pysqlite-connect) available for sqlite, each with their own protocol portion of the URI. We would need multiple checks, or `make_url(uri).drivername.startswith("sqlite"). Any sqlite URI will start with `"sqlite"`, however, so I think this way is simpler.